src/Resources/contao/classes/Theme.php

```diff
@@ -712,10 +712,8 @@ public function exportTheme(DataContainer $dc)
 		}
 
 		// Romanize the name
-		$strName = Utf8::toAscii($objTheme->name);
-		$strName = strtolower(str_replace(' ', '_', $strName));
-		$strName = preg_replace('/[^A-Za-z0-9._-]/', '', $strName);
-		$strName = basename($strName);
+		$strName = \System::getContainer()->get('contao.slugify')->slugify($objTheme->name);
+		$strName = str_replace(array('-/', '/-', '.-', '.'), array('', '', '-', ''), $strName);
```

Comment:
This is nasty. We need the service to be aware of `Config::get('folderUrl')`, which means we first have to move the setting into the application configuration.